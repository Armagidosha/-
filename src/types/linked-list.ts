import { ReactElement } from "react"
import { ElementStates } from "./element-states"

export type AnimQueue = {
  index: number | null
  showCircle: boolean
  element: ReactElement | null
  isHead?: boolean
}

export type Loading = {
  [key: string]: boolean
}