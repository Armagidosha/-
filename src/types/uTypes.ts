import { ElementStates } from "./element-states";

export type StateArray<Type> = {
  element: Type
  state: ElementStates
}