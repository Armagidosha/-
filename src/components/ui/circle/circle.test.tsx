import { Circle } from "./circle";
import renderer from "react-test-renderer"
import { ElementStates } from "../../../types/element-states";

describe("circle component", () => {

  test("circle without letters", () => {
    const circle = renderer
    .create(<Circle />)
    .toJSON()
    expect(circle).toMatchSnapshot()
  })

  test("circle with letters", () => {
    const circle = renderer
    .create(<Circle letter="abc"/>)
    .toJSON()
    expect(circle).toMatchSnapshot()
  })

  test("circle with head", () => {
    const circle = renderer
    .create(<Circle head="head"/>)
    .toJSON()
    expect(circle).toMatchSnapshot()
  })

  test("circle with head & props at the head", () => {
    const circle = renderer
    .create(<Circle head={<Circle />}/>)
    .toJSON()
    expect(circle).toMatchSnapshot()
  })

  test("circle with tail", () => {
    const circle = renderer
    .create(<Circle tail="tail"/>)
    .toJSON()
    expect(circle).toMatchSnapshot()
  })

  test("circle with tail & props at the tail", () => {
    const circle = renderer
    .create(<Circle tail={<Circle />}/>)
    .toJSON()
    expect(circle).toMatchSnapshot()
  })

  test("citcle with index", () => {
    const circle = renderer
    .create(<Circle index={1} />)
    .toJSON()
    expect(circle).toMatchSnapshot()
  })
  
  test("circle with prop isSmall", () => {
    const circle = renderer
    .create(<Circle isSmall={true} />)
    .toJSON()
    expect(circle).toMatchSnapshot()
  })

  test("circle with Default state", () => {
    const circle = renderer
    .create(<Circle state={ElementStates.Default}/>)
    .toJSON()
    expect(circle).toMatchSnapshot()
  })

  test("circle with Changing state", () => {
    const circle = renderer
    .create(<Circle state={ElementStates.Changing}/>)
    .toJSON()
    expect(circle).toMatchSnapshot()
  })

  test("circle with Modified state", () => {
    const circle = renderer
    .create(<Circle state={ElementStates.Modified}/>)
    .toJSON()
    expect(circle).toMatchSnapshot()
  })
})