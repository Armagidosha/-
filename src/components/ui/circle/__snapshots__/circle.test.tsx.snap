// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circle component circle with Changing state 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle component circle with Default state 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle component circle with Modified state 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle component circle with head & props at the head 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
      data-cy="circle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle component circle with head 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle component circle with letters 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      abc
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle component circle with prop isSmall 1`] = `
<div
  className="content "
  data-cy="small"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle component circle with tail & props at the tail 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
      data-cy="circle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`circle component circle with tail 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`circle component circle without letters 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle component citcle with index 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;
